lib/galaxy/tools/data/__init__.py

```diff
@@ -26,6 +27,31 @@
 DEFAULT_TABLE_TYPE = 'tabular'
 
 
+class ToolDataPathFiles(object):
+
+    def __init__(self, tool_data_path):
+        self.tool_data_path = tool_data_path
+        self.update_time = 0
+
+    @property
+    def tool_data_path_files(self):
+        if time.time() - self.update_time > 1:
+            content = os.walk(self.tool_data_path)
+            try:
+                self._tool_data_path_files = set([os.path.join(dirpath, fn) for dirpath, _, fn_list in content for fn in fn_list if fn and fn.endswith('.loc') or fn.endswith('.loc.sample')])
```

Comment:
Yeah, you use a generator instead of creating the list first and then converting it to a set.